### PR TITLE
[Term Entry] Go: Pointers Go

### PR DESCRIPTION
**Pull request by Mayer-04**
### Description

I have created the pointers topic in the Go programming language. Essential feature that allows you to work directly with the memory of a program.

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/go/concepts/pointers/pointers.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  - 'Memory'
+  - 'Variables'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/back-end-engineer-career-path'
+  - 'paths/computer-science'
+---
+
+## Pointers
+
```
Comment (ishg-153) [inline content/go/concepts/pointers/pointers.md:18]: Please remove this as this is not necessary.

File: content/go/concepts/pointers/pointers.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+Dereference a null pointer:
+
+```go
+  var ptr *int
+  fmt.Println(*ptr) // panic: runtime error: invalid memory address or nil pointer dereference
+```
+
+### Practical Example
```
Comment (ishg-153) [inline content/go/concepts/pointers/pointers.md:70]:
```suggestion
## Example
```

File: content/go/concepts/pointers/pointers.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+In Go, a pointer is declared using the `*` operator before the type of data to indicate that the variable is a pointer to that type of data. Then, to assign it the memory address of another variable, we use the `&` operator.
+
+```go
+var x int = 7
+var ptr *int = &x
+
+fmt.Println(ptr) // Output: 0xc000096068
+```
+
```
Comment (ishg-153) [inline content/go/concepts/pointers/pointers.md:31]: For the above code block please write its output as follows:
The output for the above code is:
(triple backticks)shell
[the received output]
(close the triple backticks)

File: content/go/concepts/pointers/pointers.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+In this example, `ptr` is a pointer to an integer (`*int`) that stores the memory address of the variable `x`.
+
+## Accessing Value through a Pointer
+
+To access the value stored at the memory address pointed to by a pointer, you use the `*` operator (`dereferencing operator`).
+
+```go
+fmt.Println(*ptr) // Output: 7
+```
+
```
Comment (ishg-153) [inline content/go/concepts/pointers/pointers.md:41]: Do the same for this and all the other code blocks with outputs.

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@Mayer-04 Nice entry! I have suggested some changes, please make those. Let me know once you are done.

**Comment by Mayer-04** [comment]
Hello, I made the respective changes to the article, I am aware of anything else.

**Review by ishg-153 — APPROVED** [review]
@Mayer-04 Ready for second review!

**Review by Sriparno08 — APPROVED** [review]
Looks good, @Mayer-04! 👍

**Pull request merged**